### PR TITLE
chore: initialize project files

### PR DESCRIPTION
**Pull request by Salakar**
Initial PR to setup project files and CI jobs. This PR does not contain any Python code.

Notes;

 - Pylint is set to use Google style guide
 - Codecov is setup as the coverage provider, however it requires activation for the repository first before we can enable it, activate via: https://app.codecov.io/gh/firebase/firebase-functions-python (needs someone with write access to the repo I think)

cc @inlined @taeold - this is ready for your review, be good to get this one in ASAP if no issues with it - will make life easier on the next PRs since they're based on this (and we will also then have CI to verify CI is green also)

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+# lines made too long by directives to pytype.
+
+# Regexp for a line that is allowed to be longer than the limit.
+ignore-long-lines=(?x)(
+  ^\s*(\#\ )?<?https?://\S+>?$|
+  ^\s*(from\s+\S+\s+)?import\s+.+$)
+
+# Allow the body of an if to be on the same line as the test if there is no
+# else.
+single-line-if-stmt=yes
```
Comment (inlined) [inline .pylintrc:274]: Is this really pythonic? I'm no pythonista but this tends to really frustrate me in other languages.

File: .pylintrc [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+# Disable the message, report, category or checker with the given id(s). You
+# can either give multiple identifiers separated by comma (,) or put this
+# option multiple times (only on the command line, not in the configuration
+# file where it should appear only once).You can also use "--disable=all" to
+# disable everything first and then reenable specific checks. For example, if
+# you want to run only the similarities checker, you can use "--disable=all
+# --enable=similarities". If you want to run only the classes checker, but have
+# no Warning level messages displayed, use"--disable=all --enable=classes
+# --disable=W"
+disable=abstract-method,
```
Comment (inlined) [inline .pylintrc:53]: This is a lot of linters being disabled. Why are they all bad? Is this just a typical list of ignored linters?

File: .pylintrc [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+# you should avoid to define new builtins when possible.
+additional-builtins=
+
+# List of strings which can identify a callback function by name. A callback
+# name must start or end with one of those strings.
+callbacks=cb_,_cb
+
+# List of qualified module names which can have objects that can redefine
+# builtins.
+redefining-builtins-modules=six,six.moves,past.builtins,future.builtins,functools
```
Comment (inlined) [inline .pylintrc:324]: Do we really need six if we're not supporting python 2?

File: mypy.ini [hunk 4]
```diff
@@ -0,0 +1,5 @@
+[mypy]
+exclude = build|setup.py
+
+[mypy-yaml.*]
+ignore_missing_imports = True
```
Comment (inlined) [inline mypy.ini:5]: Should we add the missing newline?

Comment (inlined) [reply mypy.ini:5]: here and the next two files.

**Review by inlined — APPROVED** [review]

File: mypy.ini (hunk 4 above)
Comment (Salakar) [inline mypy.ini:5]:
```suggestion
ignore_missing_imports = True

```

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+[tool.coverage]
+    [tool.coverage.run]
+    omit = [
+        '__init__.py',
+        'tests/*',
+        '*/tests/*',
+    ]
+
+    [tool.coverage.report]
+    skip_empty = true
```
Comment (Salakar) [inline pyproject.toml:14]:
```suggestion
    skip_empty = true

```

File: pytest.ini [hunk 6]
```diff
@@ -0,0 +1,5 @@
+[pytest]
+pythonpath = src/ .
+log_cli = true
+log_cli_format = "%(asctime)s [%(levelname)8s] %(message)s (%(filename)s:%(lineno)s)"
+log_cli_date_format = "%Y-%m-%d %H:%M:%S"
```
Comment (Salakar) [inline pytest.ini:5]:
```suggestion
log_cli_date_format = "%Y-%m-%d %H:%M:%S"

```

**Pull request merged**